cmd/minikube/cmd/node_add.go

```diff
@@ -36,13 +39,24 @@ var (
 	cpNode              bool
 	workerNode          bool
 	deleteNodeOnFailure bool
+	osType              string
+	windowsVersion      string
 )
 
 var nodeAddCmd = &cobra.Command{
 	Use:   "add",
 	Short: "Adds a node to the given cluster.",
 	Long:  "Adds a node to the given cluster config, and starts it.",
 	Run: func(cmd *cobra.Command, _ []string) {
+
+		if err := validateWindowsOSVersion(windowsVersion); err != nil {
+			exit.Message(reason.Usage, "{{.err}}", out.V{"err": err})
+		}
+
+		if err := validateOS(osType); err != nil {
+			exit.Message(reason.Usage, "{{.err}}", out.V{"err": err})
+		}
```

Comment:
You're going to need more validation, for example, we're not supporting the Windows node as a control plane, so you'll have to check for the control plane flag and exit it OS is Windows and control plane is true.

Reply:
I had a suggestion from Tina Murimi on the flags; would you recommend the following approach?

Why not use the format like the one used when mounting volumes  ([Volumes | Docker Docs](https://docs.docker.com/storage/volumes/#choose-the--v-or---mount-flag))? Instead of a user having to pass an additional parameter for windows-node-version
 
Something like:
--os 'os=windows,version=<VERSION>' 
 
You'll need extra functionality to parse the string but we will not need to add a new parameter for every OS.

Reply:
That's sounds good to me, just make sure to explain it well in the flag help text and provide an example.

Reply:
Let me know how the new change to the flag input looks to you @spowelljr